reconcile/openshift_resources.py

```diff
@@ -119,107 +175,134 @@ def fetch_provider_resource(path):
     return openshift_resource
 
 
+def fetch_provider_vault_secret(name, path):
+    body = {
+        "apiVersion": "v1",
+        "kind": "Secret",
+        "type": "Opaque",
+        "metadata": {
+            "name": name
+        },
+        "data": {}
+    }
+
+    # get the fields from vault
+    raw_data = vault_client.read_all(path)
+    for k, v in raw_data.items():
+        body['data'][k] = base64.b64encode(v)
```

Comment:
I checked and this doesn't wrap the text with newlines, so that's good :)

Reply:
indeed good, i checked that as well ;)